### PR TITLE
[resolve] distro cannot resolve fragments

### PR DESCRIPTION
**Pull request by rotty3000**

### CODE REVIEW

**Comment by rotty3000** [comment]
hey @bjhargrave would like to get this into release. it's small.

**Comment by rotty3000** [comment]
@bjhargrave updated.

File: biz.aQute.resolve/test/biz/aQute/resolve/ResolveTest.java [hunk 1]
```diff
@@ -318,6 +318,64 @@ public void testResolveWithLargeDistro() throws Exception {
 		}
 	}
 
+	/**
+	 * Test if we can resolve fragment with a distro without whitelisting the
+	 * 'osgi.wiring.host' capabilities on the system resource. Should FAIL
+	 */
+	public void testResolveFragmentWithDistro_FAIL() throws Exception {
+		File f = IO.getFile("testdata/repo9/fragment.bndrun");
+		File index = IO.getFile("testdata/repo9/index.xml");
```
Comment (bjhargrave) [inline biz.aQute.resolve/test/biz/aQute/resolve/ResolveTest.java]: What is this variable for? It is not used.

File: biz.aQute.resolve/test/biz/aQute/resolve/ResolveTest.java [hunk 2]
```diff
@@ -318,6 +318,64 @@ public void testResolveWithLargeDistro() throws Exception {
...
+		}
+	}
+
+	/**
+	 * Test if we can resolve fragment with a distro when whitelisting the
+	 * 'osgi.wiring.host' capabilities on the system resource. Should NOT fail
+	 */
+	public void testResolveFragmentWithDistro() throws Exception {
+		File f = IO.getFile("testdata/repo9/fragment.bndrun");
+		File index = IO.getFile("testdata/repo9/index.xml");
```
Comment (bjhargrave) [inline biz.aQute.resolve/test/biz/aQute/resolve/ResolveTest.java]: What is this variable for? It is not used.

**Comment by rotty3000** [comment]
fixed!

File: biz.aQute.resolve/src/biz/aQute/resolve/BndrunResolveContext.java [hunk 3]
```diff
@@ -61,6 +62,7 @@
 	private static final String	BND_AUGMENT					= "bnd.augment";
 	public static final String	RUN_EFFECTIVE_INSTRUCTION	= "-resolve.effective";
 	public static final String	PROP_RESOLVE_PREFERENCES	= "-resolve.preferences";
+	public static final String	NAMESPACE_WHITELIST			= "x-whitelist";
```
Comment (bjhargrave) [inline biz.aQute.resolve/src/biz/aQute/resolve/BndrunResolveContext.java]: Definitely do NOT make this field public! It should not be part of the API.

Comment (rotty3000) [reply biz.aQute.resolve/src/biz/aQute/resolve/BndrunResolveContext.java]: right..

**Pull request merged**